File: common/src/main/java/com/ssblur/scriptor/word/subject/SelfSubject.java
package com.ssblur.scriptor.word.subject;

import com.ssblur.scriptor.api.word.Subject;
import com.ssblur.scriptor.helpers.targetable.Targetable;
import com.ssblur.scriptor.word.Spell;
import net.minecraft.core.Direction;

import java.util.List;
import java.util.concurrent.CompletableFuture;

public class SelfSubject extends Subject {
  @Override
  public Cost cost() { return new Cost(1, COSTTYPE.ADDITIVE); }

  @Override
  public CompletableFuture<List<Targetable>> getTargets(Targetable caster, Spell spell) {
    var result = new CompletableFuture<List<Targetable>>();
    result.complete(List.of(caster.getFinalTargetable().setFacing(Direction.DOWN)));
    return result;
  }
}

File: common/src/main/kotlin/com/ssblur/scriptor/word/subject/SelfSubject.kt
package com.ssblur.scriptor.word.subject

import com.ssblur.scriptor.api.word.Subject
import com.ssblur.scriptor.helpers.targetable.Targetable
import com.ssblur.scriptor.word.Spell
import java.util.concurrent.CompletableFuture

class SelfSubject: Subject() {
  override fun cost() = Cost(1.0, COSTTYPE.ADDITIVE)

  override fun getTargets(caster: Targetable, spell: Spell): CompletableFuture<List<Targetable>> {
    val result = CompletableFuture<List<Targetable>>()
    result.complete(listOf(caster.finalTargetable!!.setFacing(caster.facing)))
    return result
  }
}
